File: TransCoder_tokenized_test_set_functions/python/DETECTING_NEGATIVE_CYCLE_USING_FLOYD_WARSHALL.py
def negCyclefloydWarshall ( graph ) :
    dist = [ [ 0 for i in range ( V + 1 ) ] for j in range ( V + 1 ) ]
    for i in range ( V ) :
        for j in range ( V ) :
            dist [ i ] [ j ] = graph [ i ] [ j ]
    for k in range ( V ) :
        for i in range ( V ) :
            for j in range ( V ) :
                if ( dist [ i ] [ k ] + dist [ k ] [ j ] < dist [ i ] [ j ] ) :
                    dist [ i ] [ j ] = dist [ i ] [ k ] + dist [ k ] [ j ]
    for i in range ( V ) :
        if ( dist [ i ] [ i ] < 0 ) :
            return True
    return False

File: results/java_codex/DETECTING_NEGATIVE_CYCLE_USING_FLOYD_WARSHALL.java
bool negCyclefloydWarshall ( int graph [ ] [ ] ) {
  int dist [ ] [ ] = new int [ V ] [ V ], i, j, k;
  for ( i = 0;
  i < V;
  i ++ ) for ( j = 0;
  j < V;
  j ++ ) dist [ i ] [ j ] = graph [ i ] [ j ];
  for ( k = 0;
  k < V;
  k ++ ) {
    for ( i = 0;
    i < V;
    i ++ ) {
      for ( j = 0;
      j < V;
      j ++ ) {
        if ( dist [ i ] [ k ] + dist [ k ] [ j ] < dist [ i ] [ j ] ) dist [ i ] [ j ] = dist [ i ] [ k ] + dist [ k ] [ j ];
      }
    }
  }
  for ( i = 0;
  i < V;
  i ++ ) if ( dist [ i ] [ i ] < 0 ) return true;
  return false;
}